doc/parameter.rst

```diff
@@ -59,6 +59,18 @@ General Parameters
 
   - Feature dimension used in boosting, set to maximum dimension of the feature
 
+* ``device`` [default= ``cpu``]
+
+  .. versionadded:: 2.0.0
+
+  - Device for XGBoost to run. User can set it to one of the following values:
+
+    + ``cpu``: Use CPU.
+    + ``cuda``: Use a GPU (CUDA device).
+    + ``cuda:<ordinal>``: ``<ordinal>`` is an integer that specifies the ordinal of the GPU (which GPU do you want to use if you have more than one devices).
+    + ``gpu``: Same as ``cuda``.
```

Comment:
Absolutely, that's why we have `gpu` and `cuda`. It's equal to `cuda` "for now", in the future others can make variants.

Reply:
@napetrov Feel free to share your suggestions. ;-) Would like to get more opinions.

Reply:
I mean from API perspective users would be writing code assuming GPU device are always CUDA, but this would results in breaking change when there would be another GPU backend.

Might be it worth to point that this is not the same but just a convenient way  for selecting default GPU device - so in long-term this would result in GPU dispatching regardless of particular HW.   i.e. now this is not same as 'cuda' but is default GPU device selector although it can select only from ['cuda'].  In this way we would clearly define expectations from API and would allow extensions here

Reply:
Thank you for sharing, will change the document as suggested.